core/templates/dev/head/editor/StateResponses.js

```diff
@@ -306,6 +306,45 @@ oppia.controller('StateResponses', [
       trainingDataService.initializeTrainingData(
         explorationId, currentStateName);
     };
+    $scope.suppressDefaultAnswerGroupWarnings = function() {
+      var interactionId = $scope.getCurrentInteractionId();
+      if (interactionId === 'MultipleChoiceInput') {
+        var choices = $scope.getAnswerChoices().length;
+        var answerGroups = responsesService.getAnswerGroups();
+        // First push all choice indices into an array
+        var choiceArray = [];
+        for (var i = 0; i < choices; i++) {
+          choiceArray.push(i);
+        }
+        /* Then cycle through answerGroup rule_specs
+        and put all of them into an array */
+        var answerArray = [];
+        for (var j = 0; j < answerGroups.length; j++) {
+          /* Make sure to check if a an answerGroup rule_spec
+          has more than one rule */
+          if (answerGroups['' + j + ''].rule_specs.length > 1) {
```

Comment:
I don't understand why j is being cast to a string here and elsewhere? You're just indexing into an array...